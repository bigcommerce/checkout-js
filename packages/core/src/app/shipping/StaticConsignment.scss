@import '../ui/Base';

.staticConsignmentContainer {
    ul {
        list-style: none;
        margin: 0;
    }

    & + .staticConsignmentContainer {
        margin-top: spacing("single");
        padding-top: spacing("single");
    }
}

.staticConsignment-items {
    margin: spacing("quarter") 0;
}
