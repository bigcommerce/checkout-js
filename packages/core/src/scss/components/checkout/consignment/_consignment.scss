.consignment {
    display: flex;
    flex-direction: row;
    margin: spacing("half") 0 spacing("single");
    width: 100%;
}

.consignment-product-figure {
    padding: 0 spacing("single") 0;
    width: 25%;
}

.consignmentList {
    @include u-listBullets("none");
}

.consignment-product-body {
    display: flex;
    flex-direction: column;
    width: 100%;

    h5 {
        margin-bottom: spacing("eighth");
    }

    .form-field {
        margin-top: spacing("quarter");
    }
}
