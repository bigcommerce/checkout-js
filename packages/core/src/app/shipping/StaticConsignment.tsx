import { Cart, Consignment } from '@bigcommerce/checkout-sdk';
import React, { memo, FunctionComponent } from 'react';

import { AddressType, StaticAddress } from '../address';
import { TranslatedString } from '../locale';

import { StaticShippingOption } from './shippingOption';
import './StaticConsignment.scss';
import StaticConsignmentItemList from './StaticConsignmentItemList';

interface StaticConsignmentProps {
    consignment: Consignment;
    cart: Cart;
    compactView?: boolean;
}

const StaticConsignment: FunctionComponent<StaticConsignmentProps> = ({
    consignment,
    cart,
    compactView,
}) => {
    const {
        shippingAddress: address,
        selectedShippingOption,
    } = consignment;

    return (
        <div className="staticConsignment">
            { !compactView &&
                <strong>
                    <TranslatedString id="shipping.shipping_address_heading" />
                </strong> }

            <StaticAddress
                address={ address }
                type={ AddressType.Shipping }
            />

            { !compactView &&
                <StaticConsignmentItemList
                    cart={ cart }
                    consignment={ consignment }
                /> }

            { selectedShippingOption &&
                <div>
                    { !compactView &&
                        <strong>
                            <TranslatedString id="shipping.shipping_method_label" />
                        </strong> }
                    <div className="shippingOption shippingOption--alt">
                        <StaticShippingOption
                            displayAdditionalInformation={ false }
                            method={ selectedShippingOption }
                        />
                    </div>
                </div> }
        </div>
    );
};

export default memo(StaticConsignment);
