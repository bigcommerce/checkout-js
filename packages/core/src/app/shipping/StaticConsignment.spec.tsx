import { render } from 'enzyme';
import React from 'react';

import { getCart } from '../cart/carts.mock';

import { getConsignment } from './consignment.mock';
import StaticConsignment from './StaticConsignment';

describe('StaticConsignment Component', () => {
    const consignment = getConsignment();
    const cart = getCart();

    it('renders static consignment with shipping method', () => {
        const tree = render(<StaticConsignment cart={cart} consignment={consignment} />);

        expect(tree).toMatchSnapshot();
    });

    it('renders compact view of static consignment', () => {
        const tree = render(
            <StaticConsignment cart={cart} compactView consignment={consignment} />,
        );

        expect(tree).toMatchSnapshot();
    });
});
