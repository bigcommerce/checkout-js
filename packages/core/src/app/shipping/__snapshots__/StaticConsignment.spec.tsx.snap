// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticConsignment Component renders compact view of static consignment 1`] = `
<div
  class="staticConsignment"
>
  <div>
    <div
      class="shippingOption shippingOption--alt shippingOption--selected"
    >
      <div
        class="shippingOption shippingOption--alt"
      >
        <span
          class="shippingOption-desc"
        >
          Flat Rate
        </span>
        <span
          class="shippingOption-price"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`StaticConsignment Component renders static consignment with shipping method 1`] = `
<div
  class="staticConsignment"
>
  <div
    class="staticConsignment-items"
  >
    <strong />
    <ul />
  </div>
  <div>
    <div
      class="shippingOption shippingOption--alt shippingOption--selected"
    >
      <div
        class="shippingOption shippingOption--alt"
      >
        <span
          class="shippingOption-desc"
        >
          Flat Rate
        </span>
        <span
          class="shippingOption-price"
        />
      </div>
    </div>
  </div>
</div>
`;
